AWSAuthSDK/Tests/AWSMobileClientTests/AWSMobileClientTests.swift

```diff
@@ -793,6 +795,37 @@ class AWSMobileClientTests: XCTestCase {
         
     }
     
+    func testGetIdentityId() {
+        XCTAssertNil(AWSMobileClient.sharedInstance().identityId, "Identity Id should be nil after initialize.")
+
+        let identityIdExpectation = expectation(description: "IdentityID should be retrieved successfully")
+        AWSMobileClient.sharedInstance().getIdentityId().continueWith(block: { (task) -> Any? in
+            XCTAssertNil(task.error)
+            XCTAssertNotNil(task.result, "getIdentityId should not return nil.")
+            identityIdExpectation.fulfill()
+            return nil
+        })
+        wait(for: [identityIdExpectation], timeout: 5)
+
+        guard AWSMobileClient.sharedInstance().identityId != nil else {
+            XCTFail("Could not find identityId.")
+            return
+        }
+
+        let verifyCredentialsExpectation = expectation(description: "Credentials should be retrieved successfully")
+        AWSMobileClient.sharedInstance().getAWSCredentials { (creds, error) in
+            XCTAssertNil(error)
+            XCTAssertNotNil(creds)
+            verifyCredentialsExpectation.fulfill()
+        }
+        wait(for: [verifyCredentialsExpectation], timeout: 5)
+
+        guard AWSMobileClient.sharedInstance().identityId != nil else {
```

Comment:
Generally we want to reserve `guard` for early exits. In this case, you can just `XCTAssertNotNil(AWSMobileClient.sharedInstance().identityId)`

Reply:
Changed guard to assert.